partiql-tests-data/eval/primitives/operators/overlaps.ion

```diff
@@ -0,0 +1,779 @@
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:false
+    }
+  },
+  {
+    name:"overlaps with mixed ROW and array constructors",
+    statement:"(DATE '2023-01-01', DATE '2023-01-31') OVERLAPS [DATE '2023-01-15', DATE '2023-02-15']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with null as first argument",
+    statement:"null OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with null as first argument",
+    statement:"null OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
+  {
+    name:"overlaps with null as second argument",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS null",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with null as second argument",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS null",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
+  {
+    name:"overlaps with missing as first argument",
+    statement:"MISSING OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with missing as first argument",
+    statement:"MISSING OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
+  {
+    name:"overlaps with missing as second argument",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS MISSING",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with missing as second argument",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS MISSING",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
+  {
+    name:"overlaps with string as first argument",
+    statement:"'not a datetime' OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with string as first argument",
+    statement:"'not a datetime' OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
```

Comment:
Cases like this test are a data type mismatch and should
- return MISSING in permissive / coerce mode
- give an error in strict / error mode